wealth-management-client/src/app/modal/mystock.ts

```diff
@@ -0,0 +1,27 @@
+export class MyStock {
```

Comment:
Don't need another modal object.

Reply:
`export class Stock {

  /* form data */
  script: string;
  name: string;
  investPrice: number;
  rcomPrice: number;
  targetPrice: number;
  marketPrice: number;
  rcomDate: number;
  rcomTimeScale: string;
  rcomBy: string;

  /* calculated values */
  isUserPortfolio: boolean = false;
  targetPercentage: number;
  achivedPercentage: number;
  achivedPercentageClass: string;
  achivedPercentageWidth: number;
  days: string;

  constructor() {

  }

  setMarketPrice(marketPrice: number) : void {
    this.marketPrice = marketPrice;
  }

  setGridData(data: any): void {
    var firstDate = new Date(data.rcomDate);
    var secondDate = new Date();
    this.script = data.script;
    this.name = data.name;
    this.investPrice = data.buyPrice;
    this.rcomPrice = data.rcomPrice;
    this.targetPrice = data.targetPrice;
    this.targetPercentage = (((this.targetPrice - this.rcomPrice) / this.rcomPrice) * 100);
    if (this.marketPrice !== undefined && this.marketPrice != 0) {
      if (this.isUserPortfolio) {
        this.achivedPercentage = (((this.marketPrice - this.investPrice) / this.investPrice) * 100);
        this.achivedPercentageWidth = Math.abs(this.achivedPercentage);
      } else {
        this.achivedPercentage = (((this.marketPrice - this.rcomPrice) / this.rcomPrice) * 100);
        this.achivedPercentageWidth = Math.abs(this.achivedPercentage);
      }
      this.achivedPercentageClass = (this.achivedPercentage) > 0 ? 'success' : 'danger';
    }
    this.rcomDate = new Date(data.rcomDate).getTime();
    this.days = Math.abs((firstDate.valueOf() - secondDate.valueOf()) / (24 * 60 * 60 * 1000)).toFixed(0);
    this.rcomTimeScale = data.rcomTimeScale;
    this.rcomBy = data.rcomBy;
  }
}
`

Reply:
We need other model.
We cannot mix targetPercentage and achivedPercentage based on the user investment.
The idea is to have 2 different view.
1. Recommendation View
2. Portfolio View